### ISSUE TITLE
Message gets lost if sent immediately after the open event

### ISSUE THREAD

**Issue by lorenzo-arena** [aiortc/aiortc#1194]
I have an application which consists of two parts, one of which is typically "the offerer", which also decides which datachannels to open, and the other waits for channels to be opened for communication. When a channel is opened on "the answerer" it has to do some operations (which are fast, but not immediate) before attaching a callback to the `message` event.

If I try to send a message right after receiving the `open` event in the offerer, like in the following snipped

```py
channel = pc.createDataChannel('my_custom_label')

@channel.on("message")
def on_message(message):
    logger.info(f"Received message with len {len(message)}, {message}")

@channel.on("open")
async def on_open():
    logger.info("Channel is open!")
    logger.info(f"Channel: sending message {my_message}")
    channel.send(my_message)
```

 the message seems to get lost. Is this expected behavior?

**Comment by MoreTore** [comment]
No, is not expected to lose the message on an open data channel.

**Comment by shahfaizanali** [comment]
@lorenzo-arena have you tried without using async?
